pkg/api/v20170701/types.go

```diff
@@ -0,0 +1,294 @@
+	FirstConsecutiveStaticIP string `json:"firstConsecutiveStaticIP,omitempty"`
+	StorageProfile           string `json:"storageProfile,omitempty"`
+
+	// subnet is internal
+	subnet string
+	// Master LB public endpoint/FQDN with port
+	// The format will be FQDN:2376
+	// Not used during PUT, returned as part of GET
+	FQDN string `json:"fqdn,omitempty"`
+}
+
+// AgentPoolProfile represents configuration of VMs running agent
+// daemons that register with the master and offer resources to
+// host applications in containers.
+type AgentPoolProfile struct {
+	Name           string `json:"name"`
+	Count          int    `json:"count"`
+	VMSize         string `json:"vmSize"`
+	OSDiskSizeGB   int    `json:"osDiskSizeGB,omitempty"`
+	DNSPrefix      string `json:"dnsPrefix"`
+	FQDN           string `json:"fqdn"`
+	Ports          []int  `json:"ports,omitempty"`
+	StorageProfile string `json:"storageProfile"`
+	VnetSubnetID   string `json:"vnetSubnetID,omitempty"`
+	// OSType is the operating system type for agents
+	// Set as nullable to support backward compat because
+	// this property was added later.
+	// If the value is null or not set, it defaulted to Linux.
+	OSType OSType `json:"osType,omitempty"`
+
+	// subnet is internal
+	subnet string
+}
+
+// KeyVaultSecrets specifies certificates to install on the pool
+// of machines from a given key vault
+// the key vault specified must have been granted read permissions to CRP
+type KeyVaultSecrets struct {
+	SourceVault       *KeyVaultID           `json:"sourceVault,omitempty"`
+	VaultCertificates []KeyVaultCertificate `json:"vaultCertificates,omitempty"`
+}
+
+// KeyVaultID specifies a key vault
+type KeyVaultID struct {
+	ID string `json:"id,omitempty"`
+}
+
+// KeyVaultCertificate specifies a certificate to install
+// On Linux, the certificate file is placed under the /var/lib/waagent directory
+// with the file name <UppercaseThumbprint>.crt for the X509 certificate file
+// and <UppercaseThumbprint>.prv for the private key. Both of these files are .pem formatted.
+// On windows the certificate will be saved in the specified store.
+type KeyVaultCertificate struct {
+	CertificateURL   string `json:"certificateUrl,omitempty"`
+	CertificateStore string `json:"certificateStore,omitempty"`
+}
+
+// OrchestratorType defines orchestrators supported by ACS
+type OrchestratorType string
+
+// OrchestratorVersion defines the version for orchestratorType
+type OrchestratorVersion string
+
+// UnmarshalText decodes OrchestratorType text, do a case insensitive comparison with
+// the defined OrchestratorType constant and set to it if they equal
+func (o *OrchestratorType) UnmarshalText(text []byte) error {
+	s := string(text)
+	switch {
+	case strings.EqualFold(s, string(DCOS)):
+		*o = DCOS
+	case strings.EqualFold(s, string(Kubernetes)):
+		*o = Kubernetes
+	case strings.EqualFold(s, string(DockerCE)):
+		*o = DockerCE
+	default:
+		return fmt.Errorf("OrchestratorType has unknown orchestrator: %s", s)
+	}
```

Comment:
We need to handle swarm here as well as it is like  50% of our user clusters and we need a good experience as users start using ACS.